### PR TITLE
Auto-Place CAD Components on PCB Top/Bottom Surfaces Using Board Thickness Awareness

### PR DESCRIPTION
**Pull request by Abse2001**
This PR introduces layer-aware Z-positioning for CAD components, ensuring models are correctly placed on the top or bottom surface of the PCB based on board thickness instead of relying on raw Z values.

Key improvements:

Automatically offsets CAD models to the correct PCB surface using layer and boardThickness.

Centralizes and reuses adjusted positioning across STL, GLTF, STEP, JSCAD, and Footprinter models.

Adds a comprehensive Panel Core Storybook test covering panelization and multiple PCB primitives.

Tightens rectangle border-radius clamping to avoid edge-case geometry artifacts.

Updates tscircuit dependency to leverage the latest core improvements.

Impact: more physically accurate 3D PCB visualization, fewer manual Z tweaks, and improved reliability for panelized boards and mixed CAD sources.

https://3d-viewer-git-fork-abse2001-main-tscircuit.vercel.app/?path=/story/panel-core-test--panel-core-test

### CODE REVIEW

File: src/CadViewerJscad.tsx [hunk 1]
```diff
@@ -46,9 +45,9 @@ export const CadViewerJscad = forwardRef<
   ) => {
     const childrenSoup = useConvertChildrenToCircuitJson(children)
     const internalCircuitJson = useMemo(() => {
-      const cj = soup ?? circuitJson
+      const cj = circuitJson
```
Comment (ShiboSoftwareDev) [inline src/CadViewerJscad.tsx]: lol

**Review by ShiboSoftwareDev — APPROVED** [review]
